File: src/linkedlist/lc_19_remove_node_end_list/remove_node_end_list.py
from listNode import ListNode
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        vals = []
        temp = head
        while temp:
            vals.append(temp.val)
            temp = temp.next
        idx = len(vals) - n
        vals.pop(idx)
        prev = None
        while vals:
            node = ListNode(vals.pop(), prev)
            prev = node
        return prev
        

File: src/linkedlist/lc_19_remove_node_end_list/RemoveNodeEndList.java
class RemoveNodeEndList {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int size = 0;
        ListNode temp = head;
        while(temp != null) {
            temp = temp.next;
            size++;
        }
        int target = size - n;
        if(target == 0) {
            return head.next;
        }
        
        int curr = 0;
        temp = head;
        ListNode temp1 = null;
        while(temp != null) {
            if (curr == target) {
                temp1.next = temp.next;
                break;
            }
            temp1 = temp;
            temp = temp.next;
            curr++;
        }
        return head;
    }
}